.github/workflows/build_deploy_rcb_be.yml

```diff
@@ -0,0 +1,143 @@
+name: Build & Deploy RecipeBox Backend
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - recipe-box-backend/**
+      - .github/workflows/build_deploy_rcb_be.yml
+  pull_request:
+    branches:
+      - main
+    paths:
+      - recipe-box-backend/**
+      - .github/workflows/build_deploy_rcb_be.yml
+
+# Azure login
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Download and setup Java
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: '21'
+
+      - name: Download and install maven
+        run: sudo apt -qq update & sudo apt -qq install maven -y
+
+      - name: Package server app using maven
+        run: mvn package -B -q -DskipTests=true # TODO: remove skipping tests
+        working-directory: recipe-box-backend
+
+      - name: Upload generated bundle as an artifact
```

Comment:
La fel, nu are sens sa faci upload in PR

Reply:
Nici aici nu vad sa fi existe conditie sa nu ruleze in PR sau doar cand se face build